docroot/modules/custom/foia_webform/src/FoiaSubmissionServiceApi.php

```diff
@@ -0,0 +1,254 @@
+    $apiUrl = $agencyComponent->get('field_submission_api')->value;
+
+    if (!UrlHelper::isValid($apiUrl, TRUE)) {
+      $apiUrl = FALSE;
+      $this->logger
+        ->error('Invalid API URL for the component %nid',
+          ['%nid' => $agencyComponent->id()]
+        );
+    }
+    $valuesToSubmit = NULL;
+
+    if ($apiUrl) {
+      $valuesToSubmit = $this->assembleRequestData($webformSubmission, $agencyComponent, $webform);
+    }
+    else {
+      $this->logger
+        ->error('Unable to submit request via the API. Missing API Submission URL for node %component',
+          [
+            '%component' => $agencyComponent->id(),
+            'link' => $agencyComponent->toLink(t('Edit Component'), 'edit-form')->toString(),
+          ]
+        );
+    }
+
+    if ($valuesToSubmit && $apiUrl) {
+      $this->submitToApi($apiUrl, $valuesToSubmit);
+    }
+  }
+
+  /**
+   * Gathers the required fields for the API request.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The FOIA form submission form values.
+   * @param \Drupal\node\NodeInterface $agencyComponent
+   *   The Agency Component node object.
+   * @param \Drupal\webform\WebformInterface $webform
+   *   The Webform node object.
+   *
+   * @return array
+   *   Return the assemble request data in an array.
+   */
+  public function assembleRequestData(WebformSubmissionInterface $webformSubmission, NodeInterface $agencyComponent, WebformInterface $webform) {
+    // Get the webform submission values.
+    $formValues = $this->getSubmissionValues($webformSubmission, $webform);
+
+    // Get the agency information.
+    $agencyInfo = $this->getAgencyInfo($agencyComponent);
+
+    $submissionValues = array_merge($formValues, $agencyInfo);
+
+    return $submissionValues;
+  }
+
+  /**
+   * Return the FOIA form submission values as an array.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The Webform submission values.
+   * @param \Drupal\webform\WebformInterface $webform
+   *   The Webform node object.
+   *
+   * @return array
+   *   Returns the submission values as an array.
+   */
+  public function getSubmissionValues(WebformSubmissionInterface $webformSubmission, WebformInterface $webform) {
+    $submissionValues = $webformSubmission->getData();
+    // If there are files attached, load the files and add the file metadata.
+    if ($webform->hasManagedFile()) {
+      $attachments = [];
+      $fileKeys = $this->getFileAttachmentElementsOnWebform($webform);
+      if ($fileKeys) {
+        foreach ($fileKeys as $key) {
+          $files = $this->getAttachmentData($submissionValues[$key]);
+          $attachments = array_merge($files, $attachments);
+          unset($submissionValues[$key]);
+        }
+      }
+      if (!empty($attachments)) {
+        $submissionValues['attachments'] = $attachments;
```

Comment:
The attachments should still be separated by field so that if we have two different file attachment fields that serve different purposes we retain that distinction.